src/services/package-manager/package-service.ts

```diff
@@ -264,40 +263,43 @@ class PackageService {
         logger.info(FileService.fileDownloadedMessage + filename);
     }
 
-    private async exportPackagesAndAssets(nodes: BatchExportNodeTransport[]): Promise<any[]> {
+    private async exportPackagesAndAssets(nodes: BatchExportNodeTransport[], versionsByNodeKey: Map<string, string[]>): Promise<any[]> {
         const zips = [];
-        const packages = nodes as ContentNodeTransport[];
-        for (const rootPackage of packages) {
-            const exportedPackage = await packageApi.exportPackage(rootPackage.key)
+        for (const rootPackage of nodes) {
+            let exportedPackage;
+            exportedPackage = await packageApi.exportPackage(rootPackage.key, rootPackage.version.version)
             zips.push({
                 data: exportedPackage,
-                packageKey: rootPackage.key
+                packageKey: rootPackage.key,
+                version: rootPackage.version.version
             });
         }
         return zips;
     }
 
-    private async exportToZip(nodes: BatchExportNodeTransport[]): Promise<void> {
-        const manifestNodes = this.exportManifestOfPackages(nodes);
-        const packageZips = await this.exportPackagesAndAssets(nodes);
+    private async exportToZip(nodes: BatchExportNodeTransport[], versionsByNodeKey: Map<string, string[]>): Promise<void> {
+        const manifestNodes = this.exportManifestOfPackages(nodes, versionsByNodeKey);
+        const packageZips = await this.exportPackagesAndAssets(nodes, versionsByNodeKey);
 
         const zip = new AdmZip();
 
         zip.addFile("manifest.yml", Buffer.from(YAML.stringify(manifestNodes), "utf8"));
         for (const packageZip of packageZips) {
-            zip.addFile(packageZip.packageKey + ".zip", packageZip.data)
+            zip.addFile(`${packageZip.packageKey}_${packageZip.version}.zip`, packageZip.data)
         }
-        zip.writeZip("export_" + uuidv4() + ".zip");
-        logger.info("Successfully exported package");
+
+        const fileName = "export_" + uuidv4() + ".zip";
+        zip.writeZip(fileName);
+        logger.info("Successfully exported package to: " + fileName);
     }
 
-    private exportManifestOfPackages(nodes: BatchExportNodeTransport[]): ManifestNodeTransport[] {
-        const manifestNodes: ManifestNodeTransport[] = [];
+    private exportManifestOfPackages(nodes: BatchExportNodeTransport[], dependencyVersionsByNodeKey: Map<string, string[]>): ManifestNodeTransport[] {
+        const manifestNodesByPackageKey = new Map<string, ManifestNodeTransport>();
```

Comment:
why is it better to have this as a map compared to a list?

Reply:
It is faster to search by key when we need to update the version of each package.